<a href="https://colab.research.google.com/github/meenusridhar/Airline/blob/react_node_sample/Meenatchi_SankarRaja.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#  Species Analysis

In [ ]:
#Import necessary libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

Loading the dataset using load_iris function

In [ ]:
from sklearn.datasets import load_iris
iris_dataset = load_iris()

Exploring the data keys exists in the iris dataset : Keys available [data, target, target_names, Description, feature_names, filename]

In [ ]:
print("Keys of iris_dataset:\ n{}".format(iris_dataset.keys()))

Keys of iris_dataset:\ ndict_keys(['data', 'target', 'target_names', 'DESCR', 'feature_names', 'filename'])


In [ ]:
print("printing the target_names: {}".format(iris_dataset["target_names"]))

printing the target_names: ['setosa' 'versicolor' 'virginica']


Available target Species are - setosa, versicolor, Virginica

In [ ]:
print("printing the feature_names: {}".format(iris_dataset["feature_names"]))

printing the feature_names: ['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']


The types of features available to categorize the species are Sepal length, Sepal Width, Petal length and Petal Width in Centimeters

In [ ]:
print("Print the Type of data present in the dictionary: {}".format(iris_dataset["data"]))

Print the Type of data present in the dictionary: [[5.1 3.5 1.4 0.2]
 [4.9 3.  1.4 0.2]
 [4.7 3.2 1.3 0.2]
 [4.6 3.1 1.5 0.2]
 [5.  3.6 1.4 0.2]
 [5.4 3.9 1.7 0.4]
 [4.6 3.4 1.4 0.3]
 [5.  3.4 1.5 0.2]
 [4.4 2.9 1.4 0.2]
 [4.9 3.1 1.5 0.1]
 [5.4 3.7 1.5 0.2]
 [4.8 3.4 1.6 0.2]
 [4.8 3.  1.4 0.1]
 [4.3 3.  1.1 0.1]
 [5.8 4.  1.2 0.2]
 [5.7 4.4 1.5 0.4]
 [5.4 3.9 1.3 0.4]
 [5.1 3.5 1.4 0.3]
 [5.7 3.8 1.7 0.3]
 [5.1 3.8 1.5 0.3]
 [5.4 3.4 1.7 0.2]
 [5.1 3.7 1.5 0.4]
 [4.6 3.6 1.  0.2]
 [5.1 3.3 1.7 0.5]
 [4.8 3.4 1.9 0.2]
 [5.  3.  1.6 0.2]
 [5.  3.4 1.6 0.4]
 [5.2 3.5 1.5 0.2]
 [5.2 3.4 1.4 0.2]
 [4.7 3.2 1.6 0.2]
 [4.8 3.1 1.6 0.2]
 [5.4 3.4 1.5 0.4]
 [5.2 4.1 1.5 0.1]
 [5.5 4.2 1.4 0.2]
 [4.9 3.1 1.5 0.2]
 [5.  3.2 1.2 0.2]
 [5.5 3.5 1.3 0.2]
 [4.9 3.6 1.4 0.1]
 [4.4 3.  1.3 0.2]
 [5.1 3.4 1.5 0.2]
 [5.  3.5 1.3 0.3]
 [4.5 2.3 1.3 0.3]
 [4.4 3.2 1.3 0.2]
 [5.  3.5 1.6 0.6]
 [5.1 3.8 1.9 0.4]
 [4.8 3.  1.4 0.3]
 [5.1 3.8 1.6 0.2]
 [4.6 3.2 1.4 0.2]
 [5.3 3.7 1.5 0.2]
 [5.  3.3 1.4 0.2]


In [ ]:
print("shape of the data: {}".format(iris_dataset["data"].shape))

shape of the data: (150, 4)


In [ ]:
# Shape of data contains 150 rows and 4 columns

In [ ]:
print("Type of target: {}".format(type(iris_dataset["target"])))

Type of target: <class 'numpy.ndarray'>


In [ ]:
print("Shape of target: {}".format(iris_dataset["target"].shape))

Shape of target: (150,)


In [ ]:
print("Lets see the target: {}".format(iris_dataset["target"]))

Lets see the target: [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 2 2 2 2 2 2 2 2 2 2 2
 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2
 2 2]


In [ ]:
# The data categorized as 0-setosa, 1-versicolor, 2-Virginica

Lets code for training the model using train_test_split


In [ ]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(iris_dataset['data'], iris_dataset['target'], random_state=0)

Lets print the shape of Test and train data 

In [ ]:
print("Shape of X_train:{}".format(X_train.shape))
print("Shape of y_train:{}".format(y_train.shape))
print("Shape of X_test:{}".format(X_test.shape))
print("Shape of y_test:{}".format(y_test.shape))

Shape of X_train:(112, 4)
Shape of y_train:(112,)
Shape of X_test:(38, 4)
Shape of y_test:(38,)


Out of 150 samples 112 species has been considered as training data and 38 rows has been taken for test data. The model which we are going to try is KNeighborsClassifier

In [ ]:
from sklearn.neighbors import KNeighborsClassifier
KNN =KNeighborsClassifier(n_neighbors=1)

Lets fit the model

In [ ]:
KNN.fit(X_train,y_train)

KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
                     metric_params=None, n_jobs=None, n_neighbors=1, p=2,
                     weights='uniform')

Model Prediction

In [ ]:
y_prediction = KNN.predict(X_test)
print("Test set prediction: {}".format(y_prediction))

Test set prediction: [2 1 0 2 0 2 0 1 1 1 2 1 1 1 1 0 1 1 0 0 2 1 0 0 2 0 0 1 1 0 2 1 0 2 2 1 0
 2]


Lets print the prediction score what the model predicted

In [ ]:
print("Test Score of the Model: {}".format(np.mean(y_prediction==y_test)))

Test Score of the Model: 0.9736842105263158


The model score is 97.3
